test/test_manifest_providers.py

```diff
@@ -46,9 +49,46 @@ def test_git_source():
     assert '' == package_path
     assert '<version>0.5.7</version>' in package_xml
 
-
+# mock_get_url_contents is used to mock out the '_get_url_contents' method in
+# the rosdistro.manifest_provider.github module.  Instead of going out to github
+# to get data (which can get rate-limited in Travis), it instead pulls a canned
+# example from some local files.
+def mock_get_url_contents(req):
+    import re
+    import sys
+
+    # For python3, look for the 'str' type; for Python 2, the 'unicode' type
+    if sys.version_info[0] == 3:
+        text_type = str
+    else:
+        text_type = unicode
+
+    # The urlopen() function from urllib or urllib2 takes either a string or a
+    # urllib.Request object in; determine the URL in either case.
+    if isinstance(req, text_type):
+        haystack = req
+    else:
+        haystack = req.get_full_url()
+
+    # Now see if the URL ends with 'package.xml'.  If it does, open up the file
+    # that has the canned package.xml data we have for genmsg.  Otherwise,
+    # return the canned JSON tree data that github would have returned.
+    # In both cases, store the file-like object as a module property so that we
+    # can properly close it during 'unmock_urlopen'.
+    if re.search('.*package.xml$', haystack) is not None:
+        fname = 'test/github-genmsg-package.xml'
+    else:
+        fname = 'test/github-tree-data.json'
```

Comment:
I might have just done this as a dict, but your approach may be more robust going forward.